src/planner/macos/profile_queries.rs

```diff
@@ -0,0 +1,123 @@
+use crate::planner::macos::profiles::{
+    HardDiskInternalOpts, MountControls, Policies, Profile, ProfileItem, SystemUIServer, Target,
+};
+
+struct TargetProfileItem<'a> {
+    target: &'a Target,
+    profile: &'a Profile,
+    item: &'a ProfileItem,
+}
+
+pub struct TargetProfileHardDiskInternalOpts<'a> {
+    pub target: &'a Target,
+    pub profile: &'a Profile,
+    pub opts: &'a Vec<HardDiskInternalOpts>,
+}
+
+impl TargetProfileHardDiskInternalOpts<'_> {
+    pub fn display(&self) -> String {
+        let owner = match self.target {
+            crate::planner::macos::profiles::Target::Computer => {
+                "A computer-wide profile".to_string()
+            },
+            crate::planner::macos::profiles::Target::User(u) => format!("A profile owned by {u}"),
+        };
+
+        let desc = [
+            ("Name", &self.profile.profile_display_name),
+            (
+                "Version",
+                &self.profile.profile_version.map(|v| v.to_string()),
+            ),
+            ("Description", &self.profile.profile_description),
+            ("ID", &self.profile.profile_identifier),
+            ("UUID", &self.profile.profile_uuid),
+            ("Installation Date", &self.profile.profile_install_date),
+        ]
+        .into_iter()
+        .filter_map(|(k, v)| Some((k, (*v).as_ref()?)))
+        .map(|(key, val)| format!(" * {}: {}", key, val))
+        .collect::<Vec<String>>()
+        .join("\n");
+
+        format!("{owner}:\n{}\n", desc)
+    }
+}
+
+fn flatten(policies: &Policies) -> impl Iterator<Item = TargetProfileItem> {
+    policies
+        .iter()
+        .flat_map(|(target, profiles): (&Target, &Vec<Profile>)| {
+            profiles.iter().map(move |profile| (target, profile))
+        })
+        .flat_map(|(target, profile): (&Target, &Profile)| {
+            profile
+                .profile_items
+                .iter()
+                .map(move |item| TargetProfileItem {
+                    target,
+                    profile,
+                    item,
+                })
+        })
+}
+
+pub fn blocks_internal_mounting(policies: &Policies) -> Vec<TargetProfileHardDiskInternalOpts> {
+    flatten(policies)
+        .filter_map(
+            move |TargetProfileItem {
+                      target,
+                      profile,
+                      item,
+                  }| match item {
+                ProfileItem::SystemUIServer(SystemUIServer {
+                    mount_controls: Some(MountControls { harddisk_internal }),
+                }) if harddisk_internal.iter().any(|x| {
```

Comment:
Holy guacamole this is hard to read... Can we please write this without all the destructuring, even if it means more branching?

Reply:
Is this better? lol.